src/main/java/com/app/onlinebookstore/service/impl/CartItemServiceImpl.java

```diff
@@ -66,7 +67,16 @@ public List<CartItemDto> findCartItemsByShoppingCarts(Long id) {
 
     @Override
     public void delete(Long id) {
-        cartItemRepository.deleteById(id);
+        Long userId = userService.getAuthenticatedUser().getId();

```

Comment:
This method can be significantly simplified. I suggest to make changes in cartItem repository to delete cartItems which belongs to the user. This approach will reduce number of database requests to one and improve code readability